### ISSUE TITLE
npm install fails

### ISSUE THREAD

**Issue by lookfirst** [aspnet/SignalR#794]
In the readme for the project, it gives this command: `npm install @aspnet/signalr-client --registry https://dotnet.myget.org/f/aspnetcore-ci-dev/npm/`

```
npm install @aspnet/signalr-client --registry https://dotnet.myget.org/f/aspnetcore-ci-dev/npm/
npm ERR! code E404
npm ERR! 404 Not Found: msgpack5@^3.5.0

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jon/.npm/_logs/2017-08-30T07_59_09_801Z-debug.log
```

Can you please just publish this to the normal npm repository like everyone else? It doesn't appear that `yarn` supports `--registry` either.

**Comment by davidfowl** [comment]
> @lookfirst Can you please just publish this to the normal npm repository like everyone else? It doesn't appear that yarn supports --registry either.

Nope. Not until we ship an official pre-release

**Comment by lookfirst** [comment]
Why not use the tagged release feature?

https://medium.com/@mbostock/prereleases-and-npm-e778fc5e2420

**Comment by davidfowl** [comment]
Do people put ci builds on npm?

**Comment by lookfirst** [comment]
https://docs.travis-ci.com/user/build-stages/deploy-npm/
https://docs.travis-ci.com/user/deployment/npm/

**Comment by davidfowl** [comment]
Thats not what I was asking, I was asking if it is normal to publish every single ci builds of your package to npm.

**Comment by lookfirst** [comment]
Let me bing that for you?
http://www.bing.com/search?q=I+was+asking+if+it+is+normal+to+publish+every+single+ci+builds+of+your+package+to+npm

The point of my original response is that one of the largest companies for
doing CI builds (especially for OSS work) has a documented methodology for
publishing to npm, then people must be doing it. I don't understand your
hesitation. How does the time between builds affect anything? 1.0 today.
1.0.1 an hour from now. Who cares?  I'll use semver and lock files to
control what I install.

Even still, you could simply setup a branch that you push to that is the
one that gets released. Nobody commits to that branch. You only do merges
there. Now you have CI on another branch (master) and when you want to do a
release, you push it to the 'release' branch, which does a CI build that
includes the npm publish.

p.s. Release early, release often.

jon

**Comment by moozzyk** [comment]
One of the reasons we did not push to npm was that we did not have a good naming strategy and the original name we used was taken (by our own signalr-client (sic!)). Also the msgpack5 dependency only came in last week.

The update to the README was/is in PR: https://github.com/aspnet/SignalR/pull/786

**Comment by lookfirst** [comment]
Thanks for the update, maybe you should open another issue about the naming then. ;-)

**Comment by moozzyk** [comment]
We put the client to the @aspnet scope which resolved the conflict.
